apps/passport-client/new-components/screens/Home/NewHomeScreen.tsx

```diff
@@ -0,0 +1,456 @@
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState
+} from "react";
+import { useNavigate } from "react-router-dom";
+import styled, { FlattenSimpleInterpolation, css } from "styled-components";
+import { AppContainer } from "../../../components/shared/AppContainer";
+import { CardBody } from "../../../components/shared/PCDCard";
+import { useDispatch, usePCDs, useSelf } from "../../../src/appHooks";
+import { useSyncE2EEStorage } from "../../../src/useSyncE2EEStorage";
+import { FloatingMenu } from "../../shared/FloatingMenu";
+import { NewModals } from "../../shared/Modals/NewModals";
+import { TicketCard } from "../../shared/TicketCard";
+import { Typography } from "../../shared/Typography";
+import { TicketPack, TicketType, TicketTypeName } from "./types";
+import { AddOnsModal } from "./AddOnModal";
+
+const EVENT_GAP = 4;
+const TICKETS_HORIZONTAL_GAP = 40;
+const SHOW_HELPER_LINES = false;
+const TICKET_VERTICAL_GAP = 20;
+
+const isEventTicketPCD = (pcd: PCD<unknown, unknown>): pcd is TicketType => {
+  // TODO: fetch the pods type as well and prioritize it if theres a conflict.
+  return isEdDSATicketPCD(pcd) || isPODTicketPCD(pcd);
+};
+
+const useTickets = (): Array<[string, TicketPack[]]> => {
+  const allPCDs = usePCDs();
+  const tickets = allPCDs.filter(isEventTicketPCD);
+  const ticketsTrigger = tickets.map((t) => t.id).join(" ");
+  return useMemo(() => {
+    const eventsMap = new Map<string, TicketPack[]>();
+    for (const ticket of tickets) {
+      if (ticket.claim.ticket.ticketName !== "GA") continue;
+      let ticketPacks = eventsMap.get(ticket.claim.ticket.eventName);
+      if (!ticketPacks) {
+        ticketPacks = [];
+        eventsMap.set(ticket.claim.ticket.eventName, ticketPacks);
+      }
+      if (ticket.type === PODTicketPCDTypeName) {
+        const relatedEddesaTicketPackIdx = ticketPacks.findIndex(
+          (pack) =>
+            pack.attendeeEmail === ticket.claim.ticket.attendeeEmail &&
+            pack.packType === EdDSATicketPCDTypeName
+        );
+        if (relatedEddesaTicketPackIdx >= 0)
+          ticketPacks.splice(relatedEddesaTicketPackIdx, 1);
+      }
+      ticketPacks.push({
+        eventTicket: ticket,
+        eventId: ticket.claim.ticket.eventId,
+        addOns: [],
+        attendeeEmail: ticket.claim.ticket.attendeeEmail,
+        packType: ticket.type as TicketTypeName
+      });
+    }
+    for (const ticket of tickets) {
+      if (ticket.claim.ticket.ticketName === "GA") continue;
+      const ticketPacks = eventsMap.get(ticket.claim.ticket.eventName);
+      if (!ticketPacks) continue;
+      const pack = ticketPacks.find(
+        (pack) =>
+          pack.attendeeEmail === ticket.claim.ticket.attendeeEmail &&
+          pack.packType === ticket.type
+      );
+
+      if (!pack) continue;
+      pack.addOns.push(ticket);
+    }
+    return Array.from(eventsMap.entries());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [ticketsTrigger]);
```

Comment:
generally we avoid disabling this lint unless there is a *very* good reason - could we figure out how to do this without disabling it? in vs-code you can generally auto-fix this type of lint error by clicking `cmd+.` on the red squiggle and resolving it via the dropdown that appears.

Reply:
I this logic could be causing https://www.notion.so/0xparc/ivan-s-feedback-on-PR-11271e0a5420806e97acc7fe5ed3ed60?pvs=4#11371e0a542080178a0dfd963fdf0221

Reply:
please either address this or explain why it's necessary